torch/nn/parallel/distributed.py

```diff
@@ -369,7 +379,9 @@ def _check_default_group(self):
                                "process_group argument to DDP constructor")
 
     def forward(self, *inputs, **kwargs):
-        self._sync_params()
+        if not self.accumulate_grads:
```

Comment:
It seems to me that `_sync_params` needs to be called on the first mini-batch, while `prepare_for_backward` needs to be called on the last mini-batch. Do we need an additional arg? @pietern

Reply:
@mrshenli I will leave _sync_params unchanged, because we create DistributedDataParallel per device, so the len(device_ids) = 0, we can further optimizer by not calling _sync_param in AccumulateGradientsAcrossBatches